File: Cliente en Python/PongClient.py
import socket

 

server_ip = "127.0.0.1"  # Cambia esto a la dirección IP del servidor en la nube
server_port = 8080

 

try:
    # Crear un socket de cliente y conectar al servidor
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    client_socket.connect((server_ip, server_port))
    print("Conectado al servidor Pong.")

 

    # Lógica de comunicación con el servidor
    while True:
        # Leer entrada del usuario
        message = input("Ingrese un mensaje: ")

 

        # Enviar el mensaje al servidor
        client_socket.send(message.encode())

 

        # Recibir respuesta del servidor
        response = client_socket.recv(1024).decode()
        print("Respuesta del servidor:", response)

 

except ConnectionRefusedError:
    print("No se pudo conectar al servidor. Asegúrate de que el servidor esté en ejecución.")
except Exception as e:
    print("Error de comunicación con el servidor:", str(e))
finally:
    client_socket.close()

File: Cliente/PongClient.java
import java.io.*;
import java.net.*;

public class PongClient {
    public static void main(String[] args) {
        String serverIP = "13.58.181.109"; // Cambia esto a la dirección IP del servidor en la nube
        int serverPort = 8080;

        try {
            // Crear un socket de cliente y conectar al servidor
            Socket socket = new Socket(serverIP, serverPort);
            System.out.println("Conectado al servidor Pong.");

            // Configurar flujos de entrada y salida
            BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);

            // Lógica de comunicación con el servidor
            BufferedReader userInput = new BufferedReader(new InputStreamReader(System.in));
            String message;

            System.out.print("Ingrese su nombre de usuario: ");
            String username = userInput.readLine();
            
            // Enviar el nombre de usuario al servidor
            out.println("USERNAME:" + username);

            while (true) {
                // Leer entrada del usuario
                System.out.print("Ingrese un mensaje: ");
                message = userInput.readLine();

                // Enviar el mensaje al servidor
                out.println(message);

                // Recibir respuesta del servidor
                String response = in.readLine();
                if (response.startsWith("BEST_SCORE:")) {
                    int bestScore = Integer.parseInt(response.substring(11));
                    System.out.println("Mejor puntuación: " + bestScore);
                } else {
                    System.out.println("Respuesta del servidor: " + response);
                }
            }

        } catch (IOException e) {
            System.err.println("Error de comunicación con el servidor: " + e.getMessage());
        }
    }
}
